caliban_toolbox/build_test.py

```diff
@@ -167,100 +224,77 @@ def test_pad_image_stack():
-                                                     resize_ratios=resize_ratios,
-                                                     final_size=final_size)
-
-    # check that correct number of NPZs present
-    assert combined_x.shape[0] == np.sum(num_images)
-
-    # check correct size of NPZs
-    assert combined_x.shape[1:3] == final_size
-
-    # NPZ files need to be cropped
-    num_images = [2, 2]
-    sizes = [(512, 512), (512, 512)]
-    npz_crop_list = _make_npzs(sizes=sizes, num_images=num_images)
-    resize_ratios = [[1], [1]]
-    final_size = (256, 256)
-
-    combined_x, combined_y = build.combine_npz_files(npz_list=npz_crop_list,
-                                                     resize_ratios=resize_ratios,
-                                                     final_size=final_size)
-
-    # check that correct number of NPZs present
-    assert combined_x.shape[0] == np.sum(num_images) * 4
-
-    # check correct size of NPZs
-    assert combined_x.shape[1:3] == final_size
-
-    # NPZ files need to be resized
-    num_images = [2, 2]
-    sizes = [(128, 128), (128, 128)]
-    npz_resize_list = _make_npzs(sizes=sizes, num_images=num_images)
-    resize_ratios = [[2], [2]]
-    final_size = (256, 256)
-
-    combined_x, combined_y = build.combine_npz_files(npz_list=npz_resize_list,
-                                                     resize_ratios=resize_ratios,
-                                                     final_size=final_size)
-
-    # check that correct number of NPZs present
-    assert combined_x.shape[0] == np.sum(num_images)
-
-    # check correct size of NPZs
-    assert combined_x.shape[1:3] == final_size
-
-    # some need to be cropped, some need to be resized
-    npz_list = npz_crop_list + npz_resize_list
-    resize_ratios = [[1], [1], [2], [2]]
-    final_size = (256, 256)
-
-    combined_npz = build.combine_npz_files(npz_list=npz_list, resize_ratios=resize_ratios,
-                                           final_size=final_size)
-
-    combined_x, combined_y = combined_npz
-
-    # check that correct number of NPZs present
-    assert combined_x.shape[0] == (np.sum(num_images) + np.sum(num_images) * 4)
-
-    # check correct size of NPZs
-    assert combined_x.shape[1:3] == final_size
-
-    # different resizing for each image in the NPZ
-    num_images = [2, 2]
-    sizes = [(256, 256), (256, 256)]
-    npz_resize_list = _make_npzs(sizes=sizes, num_images=num_images)
-    resize_ratios = [[1], [1, 2]]
-    final_size = (256, 256)
-
-    combined_x, combined_y = build.combine_npz_files(npz_list=npz_resize_list,
-                                                     resize_ratios=resize_ratios,
-                                                     final_size=final_size)
-
-    # check that correct number of NPZs present
-    assert combined_x.shape[0] == np.sum(2 + 1 + 4)
+def test_train_val_test_split():
+    X_data = np.zeros((100, 5, 5, 3))
+    y_data = np.zeros((100, 5, 5, 1))
```

Comment:
You may want to add another test that confirms reproducibility with a given seed.